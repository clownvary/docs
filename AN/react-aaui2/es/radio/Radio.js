import _extends from 'babel-runtime/helpers/extends';
import _objectWithoutProperties from 'babel-runtime/helpers/objectWithoutProperties';
import _Object$defineProperties from 'babel-runtime/core-js/object/define-properties';
import _classCallCheck from 'babel-runtime/helpers/classCallCheck';
import _possibleConstructorReturn from 'babel-runtime/helpers/possibleConstructorReturn';
import _inherits from 'babel-runtime/helpers/inherits';
import React, { Component } from 'react';
import { bool, number, string, object, func, node, oneOfType } from 'prop-types';

import RadioComponent from './RadioComponent';
import { identity } from '../shared/utils';

var Radio = function (_Component) {
  _inherits(Radio, _Component);

  function Radio() {
    var _temp, _this, _ret;

    _classCallCheck(this, Radio);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, _Component.call.apply(_Component, [this].concat(args))), _this), _this.setRef = function (radioComponent) {
      _this.radioComponent = radioComponent;
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  // Support legacy API for `this.refs.checkbox.checked`
  // It's not recommended because it breaks component encapsulation and
  // it violates the unidirectional data flow for React
  Radio.prototype.componentDidMount = function componentDidMount() {
    _Object$defineProperties(this, {
      checked: {
        get: function get() {
          return this.radioComponent.checked;
        },
        set: function set(v) {
          this.radioComponent.checked = !!v;
        }
      }
    });
  };

  Radio.prototype.render = function render() {
    var _props = this.props,
        children = _props.children,
        rest = _objectWithoutProperties(_props, ['children']);

    var auiRadioGroup = this.context.auiRadioGroup;

    // Proxy `onChange`, `checked` and `disabled` to `auiRadioGroup`

    if (auiRadioGroup) {
      rest.onChange = auiRadioGroup.onChange;
      rest.checked = this.props.value === auiRadioGroup.value;
      rest.disabled = this.props.disabled || auiRadioGroup.disabled;
      rest.size = this.props.size || auiRadioGroup.size;
      rest.name = this.props.name || auiRadioGroup.name;
    }

    return React.createElement(
      RadioComponent,
      _extends({ ref: this.setRef }, rest),
      children
    );
  };

  return Radio;
}(Component);

Radio.displayName = 'AUIRadio';
Radio.propTypes = {
  value: oneOfType([string, number]),
  disabled: bool,
  size: string,
  name: string,
  children: node,
  onChange: func
};
Radio.defaultProps = {
  onChange: identity
};
Radio.contextTypes = {
  auiRadioGroup: object };
export default Radio;