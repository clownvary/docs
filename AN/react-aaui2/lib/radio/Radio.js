'use strict';

exports.__esModule = true;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _defineProperties = require('babel-runtime/core-js/object/define-properties');

var _defineProperties2 = _interopRequireDefault(_defineProperties);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _RadioComponent = require('./RadioComponent');

var _RadioComponent2 = _interopRequireDefault(_RadioComponent);

var _utils = require('../shared/utils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Radio = function (_Component) {
  (0, _inherits3.default)(Radio, _Component);

  function Radio() {
    var _temp, _this, _ret;

    (0, _classCallCheck3.default)(this, Radio);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, _Component.call.apply(_Component, [this].concat(args))), _this), _this.setRef = function (radioComponent) {
      _this.radioComponent = radioComponent;
    }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);
  }

  // Support legacy API for `this.refs.checkbox.checked`
  // It's not recommended because it breaks component encapsulation and
  // it violates the unidirectional data flow for React
  Radio.prototype.componentDidMount = function componentDidMount() {
    (0, _defineProperties2.default)(this, {
      checked: {
        get: function get() {
          return this.radioComponent.checked;
        },
        set: function set(v) {
          this.radioComponent.checked = !!v;
        }
      }
    });
  };

  Radio.prototype.render = function render() {
    var _props = this.props,
        children = _props.children,
        rest = (0, _objectWithoutProperties3.default)(_props, ['children']);
    var auiRadioGroup = this.context.auiRadioGroup;

    // Proxy `onChange`, `checked` and `disabled` to `auiRadioGroup`

    if (auiRadioGroup) {
      rest.onChange = auiRadioGroup.onChange;
      rest.checked = this.props.value === auiRadioGroup.value;
      rest.disabled = this.props.disabled || auiRadioGroup.disabled;
      rest.size = this.props.size || auiRadioGroup.size;
      rest.name = this.props.name || auiRadioGroup.name;
    }

    return _react2.default.createElement(
      _RadioComponent2.default,
      (0, _extends3.default)({ ref: this.setRef }, rest),
      children
    );
  };

  return Radio;
}(_react.Component);

Radio.displayName = 'AUIRadio';
Radio.propTypes = {
  value: (0, _propTypes.oneOfType)([_propTypes.string, _propTypes.number]),
  disabled: _propTypes.bool,
  size: _propTypes.string,
  name: _propTypes.string,
  children: _propTypes.node,
  onChange: _propTypes.func
};
Radio.defaultProps = {
  onChange: _utils.identity
};
Radio.contextTypes = {
  auiRadioGroup: _propTypes.object };
exports.default = Radio;
module.exports = exports['default'];