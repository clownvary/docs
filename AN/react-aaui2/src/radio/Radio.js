import React, { Component } from 'react'
import { bool, number, string, object, func, node, oneOfType } from 'prop-types'

import RadioComponent from './RadioComponent'
import { identity } from '../shared/utils'

export default class Radio extends Component {
  static displayName = 'AUIRadio'

  static propTypes = {
    value: oneOfType([string, number]),
    disabled: bool,
    size: string,
    name: string,
    children: node,
    onChange: func,
  }

  static defaultProps = {
    onChange: identity,
  }

  static contextTypes = {
    auiRadioGroup: object,
  }

  // Support legacy API for `this.refs.checkbox.checked`
  // It's not recommended because it breaks component encapsulation and
  // it violates the unidirectional data flow for React
  componentDidMount() {
    Object.defineProperties(this, {
      checked: {
        get() {
          return this.radioComponent.checked
        },
        set(v) {
          this.radioComponent.checked = !!v
        },
      },
    })
  }

  setRef = radioComponent => {
    this.radioComponent = radioComponent
  }

  render() {
    const { children, ...rest } = this.props
    const { auiRadioGroup } = this.context

    // Proxy `onChange`, `checked` and `disabled` to `auiRadioGroup`
    if (auiRadioGroup) {
      rest.onChange = auiRadioGroup.onChange
      rest.checked = this.props.value === auiRadioGroup.value
      rest.disabled = this.props.disabled || auiRadioGroup.disabled
      rest.size = this.props.size || auiRadioGroup.size
      rest.name = this.props.name || auiRadioGroup.name
    }

    return (
      <RadioComponent ref={this.setRef} {...rest}>
        {children}
      </RadioComponent>
    )
  }
}
