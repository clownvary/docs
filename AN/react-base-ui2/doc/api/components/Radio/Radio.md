## Radio 
---
UI component of Radio

### Prop Types
Property | Type | Default Value | Is Required | Description
:--- | :--- | :--- | :--- | :---
value|string &#124; number|&ensp;|false|Radio value.
disabled|bool|&ensp;|false|Determines enable/disable state.
size|string|&ensp;|false|Determines the Radio size.
name|string|&ensp;|false|Field name.
children|node|&ensp;|false|Child nodes.
onChange|func|identity|false|Fires when value change.