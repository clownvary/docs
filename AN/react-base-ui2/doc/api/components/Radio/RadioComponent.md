## RadioComponent 



### Prop Types
Property | Type | Default Value | Is Required | Description
:--- | :--- | :--- | :--- | :---
disabled|bool|&ensp;|false|&ensp;
checked|bool|&ensp;|false|&ensp;
defaultChecked|bool|false|false|&ensp;
className|string|&ensp;|false|&ensp;
style|object|&ensp;|false|&ensp;
onChange|func|identity|false|&ensp;
children|node|&ensp;|false|&ensp;
size|enum:<br>['sm', 'md', 'lg']|Size3.MEDIUM|false|&ensp;
value|any|&ensp;|false|&ensp;
ariaLabel|string|&ensp;|false|&ensp;