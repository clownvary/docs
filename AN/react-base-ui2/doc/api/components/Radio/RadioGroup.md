## RadioGroup 
---
UI component that displays a group of Radio.

### Prop Types
Property | Type | Default Value | Is Required | Description
:--- | :--- | :--- | :--- | :---
disabled|bool|&ensp;|false|Determines the enable/disable state.
size|string|&ensp;|false|Determines the Radio size.
name|string|&ensp;|false|Field name.
className|string|&ensp;|false|Customize CSS class name.
onChange|func|identity|false|Fires when value change.
data|[{<br>&emsp;&emsp;text: string,<br>&emsp;&emsp;value: any<br>}]|&ensp;|false|Array of child items. Each item is a text/value pair.
children|node|&ensp;|false|Child node
value|any|&ensp;|false|Value of radio group.
defaultValue|any|&ensp;|false|Radio group value in default state.