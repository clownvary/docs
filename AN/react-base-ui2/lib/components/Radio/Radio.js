'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _extends2 = require('babel-runtime/helpers/extends');var _extends3 = _interopRequireDefault(_extends2);var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);var _defineProperties = require('babel-runtime/core-js/object/define-properties');var _defineProperties2 = _interopRequireDefault(_defineProperties);var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);var _createClass2 = require('babel-runtime/helpers/createClass');var _createClass3 = _interopRequireDefault(_createClass2);var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);var _inherits2 = require('babel-runtime/helpers/inherits');var _inherits3 = _interopRequireDefault(_inherits2);var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _propTypes = require('prop-types');
var _identity = require('lodash/identity');var _identity2 = _interopRequireDefault(_identity);

var _RadioComponent = require('./RadioComponent');var _RadioComponent2 = _interopRequireDefault(_RadioComponent);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

/** Default PropTypes of Radio.
                                                                                                                                                                                                                * @memberof Radio
                                                                                                                                                                                                               */
var RadioPropTypes = {
  /** Radio value.
                        * @type {string|number}
                        */
  value: (0, _propTypes.oneOfType)([_propTypes.string, _propTypes.number]),
  /** Determines enable/disable state.
                                                                             * @type {boolean}
                                                                             */
  disabled: _propTypes.bool,
  /** Determines the Radio size.
                              * @type {Size}
                              */
  size: _propTypes.string,
  /** Field name.
                            * @type {string}
                            */
  name: _propTypes.string,
  /** Child nodes.
                            * @type {node}
                            */
  children: _propTypes.node,
  /**  Fires when value change.
                              * @event
                              * @type {func}
                             */
  onChange: _propTypes.func };


var RadioProps = {
  onChange: _identity2.default };


/** UI component of Radio */var
Radio = function (_Component) {(0, _inherits3.default)(Radio, _Component);function Radio() {var _ref;var _temp, _this, _ret;(0, _classCallCheck3.default)(this, Radio);for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {args[_key] = arguments[_key];}return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, (_ref = Radio.__proto__ || (0, _getPrototypeOf2.default)(Radio)).call.apply(_ref, [this].concat(args))), _this), _this.
























    setRef = function (radioComponent) {
      _this.radioComponent = radioComponent;
    }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);}(0, _createClass3.default)(Radio, [{ key: 'componentDidMount', // Support legacy API for `this.refs.checkbox.checked`
    // It's not recommended because it breaks component encapsulation and
    // it violates the unidirectional data flow for React
    value: function componentDidMount() {(0, _defineProperties2.default)(this, { checked: { get: function get() {return this.radioComponent.checked;}, set: function set(v) {this.radioComponent.checked = !!v;} } });} }, { key: 'render', value: function render() {var _props = this.props,children = _props.children,rest = (0, _objectWithoutProperties3.default)(_props, ['children']);var
      auiRadioGroup = this.context.auiRadioGroup;

      // Proxy `onChange`, `checked` and `disabled` to `auiRadioGroup`
      if (auiRadioGroup) {
        rest.onChange = auiRadioGroup.onChange;
        rest.checked = this.props.value === auiRadioGroup.value;
        rest.disabled = this.props.disabled || auiRadioGroup.disabled;
        rest.size = this.props.size || auiRadioGroup.size;
        rest.name = this.props.name || auiRadioGroup.name;
      }

      return (
        _react2.default.createElement(_RadioComponent2.default, (0, _extends3.default)({ ref: this.setRef }, rest),
          children));


    } }]);return Radio;}(_react.Component);Radio.displayName = 'Radio';Radio.propTypes = RadioPropTypes;Radio.defaultProps = RadioProps;Radio.contextTypes = { auiRadioGroup: _propTypes.object };exports.default =


Radio;module.exports = exports['default'];