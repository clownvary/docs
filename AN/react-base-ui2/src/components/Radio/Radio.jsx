import React, { Component } from 'react';
import { bool, number, string, object, func, node, oneOfType } from 'prop-types';
import identity from 'lodash/identity';

import RadioComponent from './RadioComponent';

/** Default PropTypes of Radio.
 * @memberof Radio
*/
const RadioPropTypes = {
  /** Radio value.
   * @type {string|number}
   */
  value: oneOfType([string, number]),
  /** Determines enable/disable state.
   * @type {boolean}
   */
  disabled: bool,
  /** Determines the Radio size.
   * @type {Size}
   */
  size: string,
  /** Field name.
   * @type {string}
   */
  name: string,
  /** Child nodes.
   * @type {node}
   */
  children: node,
  /**  Fires when value change.
   * @event
   * @type {func}
  */
  onChange: func
};

const RadioProps = {
  onChange: identity
};

/** UI component of Radio */
class Radio extends Component {
  static displayName = 'Radio'
  static propTypes = RadioPropTypes;
  static defaultProps = RadioProps;

  static contextTypes = {
    auiRadioGroup: object
  }

  // Support legacy API for `this.refs.checkbox.checked`
  // It's not recommended because it breaks component encapsulation and
  // it violates the unidirectional data flow for React
  componentDidMount() {
    Object.defineProperties(this, {
      checked: {
        get() {
          return this.radioComponent.checked;
        },
        set(v) {
          this.radioComponent.checked = !!v;
        }
      }
    });
  }

  setRef = (radioComponent) => {
    this.radioComponent = radioComponent;
  }

  render() {
    const { children, ...rest } = this.props;
    const { auiRadioGroup } = this.context;

    // Proxy `onChange`, `checked` and `disabled` to `auiRadioGroup`
    if (auiRadioGroup) {
      rest.onChange = auiRadioGroup.onChange;
      rest.checked = this.props.value === auiRadioGroup.value;
      rest.disabled = this.props.disabled || auiRadioGroup.disabled;
      rest.size = this.props.size || auiRadioGroup.size;
      rest.name = this.props.name || auiRadioGroup.name;
    }

    return (
      <RadioComponent ref={this.setRef} {...rest}>
        {children}
      </RadioComponent>
    );
  }
}

export default Radio;
