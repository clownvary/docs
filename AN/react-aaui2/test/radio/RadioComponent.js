import React from 'react'
import renderer from 'react-test-renderer'
import { shallow, mount } from 'enzyme'

import Radio from 'radio/RadioComponent'

it('should render without errors', () => {
  const wrapper = shallow(<Radio />)

  expect(wrapper).toBeTruthy()
})

it('should render right thing', () => {
  const wrapper = shallow(<Radio />)

  expect(wrapper.find('label.radio-wrapper').length).toBe(1)
  expect(wrapper.find('input').length).toBe(1)
  expect(wrapper.find('span.radio__inner').length).toBe(1)
})

it("should render right thing and don't change unexpected", () => {
  const tree = renderer.create(<Radio />).toJSON()

  expect(tree).toMatchSnapshot()
})

it('has a .radio-wrapper class name for the default Radio', () => {
  const wrapper = shallow(<Radio />)

  expect(wrapper.hasClass('radio-wrapper')).toBe(true)
})

it('disaled', () => {
  const wrapper = shallow(<Radio disabled />)

  expect(wrapper.find('input[disabled]').length).toBe(1)
})

it('has a .radio and .radio--sm class name for the common size Radio', () => {
  const wrapper = shallow(<Radio size="sm" />)

  expect(wrapper.find('.radio.radio--sm').length).toBe(1)
})

it('has a .radio and .radio--lg class name for the large Radio', () => {
  const wrapper = shallow(<Radio size="lg" />)

  expect(wrapper.find('.radio.radio--lg').length).toBe(1)
})

it('should could checked through the instance property', () => {
  const wrapper = mount(<Radio defaultChecked />)

  expect(wrapper.instance().checked).toBe(true)
})

it('should could set checked through the instance `checked` property when defaultChecked', () => {
  const wrapper = mount(<Radio defaultChecked />)

  expect(wrapper.instance().checked).toBe(true)
  expect(wrapper.state('checked')).toBe(true)

  wrapper.instance().checked = false

  expect(wrapper.instance().checked).toBe(false)
  expect(wrapper.instance().input.checked).toBe(false)
  expect(wrapper.state('checked')).toBe(false)
})

it('should could set checked through the instance `checked` property when checked', () => {
  const wrapper = mount(<Radio checked />)

  expect(wrapper.instance().checked).toBe(true)
  expect(wrapper.state('checked')).toBe(true)

  wrapper.instance().checked = false

  expect(wrapper.instance().checked).toBe(false)
  expect(wrapper.instance().input.checked).toBe(false)
  expect(wrapper.state('checked')).toBe(false)
  expect(wrapper.state('checked')).toBe(false)
})

it('should could set checked through the instance `checked` property when unchecked', () => {
  const wrapper = mount(<Radio />)

  expect(wrapper.instance().checked).toBe(false)
  expect(wrapper.state('checked')).toBe(false)

  wrapper.instance().checked = true

  expect(wrapper.instance().checked).toBe(true)
  expect(wrapper.instance().input.checked).toBe(true)
  expect(wrapper.state('checked')).toBe(true)
})

it('should NOT call onChange when setting checked through the instance property', () => {
  const onChangeMock = jest.fn()
  const wrapper = mount(<Radio defaultChecked onChange={onChangeMock} />)

  wrapper.instance().checked = false

  expect(wrapper.instance().checked).toBe(false)
  expect(onChangeMock).not.toBeCalled()
})

it('should could sync props => state', () => {
  const wrapper = shallow(<Radio defaultChecked />)

  expect(wrapper.state('checked')).toBe(true)

  wrapper.setProps({
    useless: !wrapper.state('checked'),
  })

  wrapper.setProps({
    checked: !wrapper.state('checked'),
  })

  expect(wrapper.state('checked')).toBe(false)

  wrapper.setProps({
    checked: wrapper.state('checked'),
  })
})

it('should call onChange when triggering change event for the wrapped input', () => {
  const onChangeMock = jest.fn()
  const wrapper = shallow(<Radio onChange={onChangeMock} />)

  expect(wrapper.state('checked')).toBe(false)

  wrapper.find('input[type="radio"]').simulate('change', {
    persist: () => {},
    preventDefault: () => {},
    target: {
      checked: true,
    },
  })

  expect(onChangeMock).toBeCalled()
  expect(wrapper.state('checked')).toBe(true)
})

it("don't handle `onChange` event when setting `disabled` as true", () => {
  const onChangeMock = jest.fn()
  const wrapper = shallow(<Radio disabled checked onChange={onChangeMock} />)

  wrapper.find('input').simulate('change', {
    target: {
      value: false,
    },
  })

  expect(onChangeMock).not.toBeCalled()
})

it('should could call `stopPropagation` & `preventDefault`', () => {
  const stopPropagationMock = jest.fn()
  const preventDefaultMock = jest.fn()
  const onChangeMock = jest.fn(e => {
    e.stopPropagation()
    e.preventDefault()
  })
  const wrapper = shallow(<Radio onChange={onChangeMock} />)

  wrapper.find('input').simulate('change', {
    target: {
      checked: true,
    },
    stopPropagation: stopPropagationMock,
    preventDefault: preventDefaultMock,
  })

  expect(stopPropagationMock).toBeCalled()
  expect(preventDefaultMock).toBeCalled()
})
