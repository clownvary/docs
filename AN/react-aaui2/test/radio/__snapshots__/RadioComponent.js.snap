// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render right thing and don't change unexpected 1`] = `
<label
  className="radio-wrapper"
  style={undefined}
>
  <span
    className="radio"
  >
    <input
      aria-hidden={true}
      checked={false}
      className="radio__input"
      disabled={undefined}
      onChange={[Function]}
      type="radio"
    />
    <span
      aria-checked={false}
      aria-disabled={undefined}
      className="radio__inner"
      role="radio"
    />
    <span />
  </span>
</label>
`;
